File: Maximum Product Subarray/maximum product subarray.cpp
class Solution {
public:
    int maxProduct(vector<int>& nums) {
        int ans = 0;
        if (nums.size() == 1) {
            return nums[0];
        }
        
        int currentProduct = 1;
        
        for(int i = 0; i < nums.size() - 1; i++) {
            if (nums[i] != 0) {
                currentProduct = currentProduct * nums[i];
                if (currentProduct > ans) {
                    ans = currentProduct;
                }
            }
            else {
                currentProduct = 1;
            }
        }
        
        currentProduct = 1;
        for(int i = nums.size() - 1; i >= 0; i--) {
            if (nums[i] != 0) {
                currentProduct = currentProduct * nums[i];
                if (currentProduct > ans) {
                    ans = currentProduct;
                }
            }
            else {
                currentProduct = 1;
            }
        }
        
        return ans;
    }
};

File: Maximum Product Subarray/maximum product subarray.py
class Solution:
    def maxProduct(self, nums: List[int]) -> int:
        ans = 0
        
        if (len(nums) == 1):
            return nums[0]
        
        currentProduct = 1
        for elem in nums:
            if (elem != 0):
                currentProduct = currentProduct * elem
                ans = max(ans, currentProduct)
            else:
                currentProduct = 1
        
        currentProduct = 1
        for i in range(len(nums) - 1, -1, -1):
            if nums[i] != 0:
                currentProduct = currentProduct * nums[i]
                ans = max(ans, currentProduct)
            else:
                currentProduct = 1
            
        return ans